### ISSUE TITLE
[Debug] Getting error Script execution timed out

### ISSUE THREAD

**Issue by sguptatw** [getgauge/gauge-vscode#237]
**Expected behavior**
Should be able to debug seamlessly

**Actual behavior**
Getting the error `Script execution timed out`
```
    at ontimeout (timers.js:458:11)
    at VM.run (.gauge/plugins/js/2.2.0/src/vm.js:61:13)
    at VM.runFile (.gauge/plugins/js/2.2.0/src/vm.js:73:8)
    at .../impl-loader.js:9:8
    at Array.forEach (<anonymous>)
    at Object.loadImpl [as load] (.gauge/plugins/js/2.2.0/src/impl-loader.js:6:29)
    at startExecution (.../.gauge/plugins/js/2.2.0/src/message-processor.js:66:15)
    at Timeout._onTimeout (.../.gauge/plugins/js/2.2.0/src/message-processor.js:78:30)
    at ontimeout (timers.js:458:11)
    at tryOnTimeout (timers.js:296:5)
    at Timer.listOnTimeout (timers.js:259:5)
```
**Steps to replicate**
* On project 
[basic.zip](https://github.com/getgauge/gauge-vscode/files/1927933/basic.zip)
* Add debugging point on line:83 of step_implementation.js
* Start debugging
* Click on `proceed` in the debug panel
> The error appears on console

**Version**
```
Gauge version: 0.9.8
Commit Hash: 61d692c

Plugins
-------
html-report (4.0.4.nightly-2018-04-11)
js (2.2.0)
```

**Comment by sguptatw** [comment]
This is when the debug point is on the definition of beforeScenario.

When the debug point is on the definition of step definition, getting `use of network connection closed`

```
Running tool: gauge run /Users/sswaroop/work/testfolder/js/basic/specs/example.spec:13 --simple-console --hide-suggestion
Runner Ready for Debugging
Debugger listening on ws://127.0.0.1:51183/61212b68-8b0a-45b5-9b7c-4d3c4721d7cb
For help see https://nodejs.org/en/docs/inspector
Debugger attached.
Error executing tests/step_implementation.js
Trace: Error: Script execution timed out.
    at Script.runInContext (vm.js:74:29)
    at Object.runInContext (vm.js:184:6)
    at VM.run (/Users/sswaroop/.gauge/plugins/js/2.2.0/src/vm.js:58:8)
    at VM.runFile (/Users/sswaroop/.gauge/plugins/js/2.2.0/src/vm.js:73:8)
    at /Users/sswaroop/.gauge/plugins/js/2.2.0/src/impl-loader.js:9:8
    at Array.forEach (<anonymous>)
    at Object.loadImpl [as load] (/Users/sswaroop/.gauge/plugins/js/2.2.0/src/impl-loader.js:6:29)
    at startExecution (/Users/sswaroop/.gauge/plugins/js/2.2.0/src/message-processor.js:66:15)
    at Timeout._onTimeout (/Users/sswaroop/.gauge/plugins/js/2.2.0/src/message-processor.js:78:30)
    at ontimeout (timers.js:458:11)
    at VM.run (/Users/sswaroop/.gauge/plugins/js/2.2.0/src/vm.js:61:13)
    at VM.runFile (/Users/sswaroop/.gauge/plugins/js/2.2.0/src/vm.js:73:8)
    at /Users/sswaroop/.gauge/plugins/js/2.2.0/src/impl-loader.js:9:8
    at Array.forEach (<anonymous>)
    at Object.loadImpl [as load] (/Users/sswaroop/.gauge/plugins/js/2.2.0/src/impl-loader.js:6:29)
    at startExecution (/Users/sswaroop/.gauge/plugins/js/2.2.0/src/message-processor.js:66:15)
    at Timeout._onTimeout (/Users/sswaroop/.gauge/plugins/js/2.2.0/src/message-processor.js:78:30)
    at ontimeout (timers.js:458:11)
    at tryOnTimeout (timers.js:296:5)
    at Timer.listOnTimeout (timers.js:259:5)
# Specification Heading
  ## DDD
/Users/sswaroop/.gauge/plugins/js/2.2.0/src/executor.js:46
  new Test(step.fn, parameters, timeout).run().then(
                ^

TypeError: Cannot read property 'fn' of undefined
    at Object.executeStep [as step] (/Users/sswaroop/.gauge/plugins/js/2.2.0/src/executor.js:46:17)
    at MessageProcessor.executeStep (/Users/sswaroop/.gauge/plugins/js/2.2.0/src/message-processor.js:39:26)
    at MessageProcessor.getResponseFor (/Users/sswaroop/.gauge/plugins/js/2.2.0/src/message-processor.js:316:40)
    at ExecutionConnection.<anonymous> (/Users/sswaroop/.gauge/plugins/js/2.2.0/src/gauge.js:41:19)
    at ExecutionConnection.emit (events.js:160:13)
    at Socket.messageHandler (/Users/sswaroop/.gauge/plugins/js/2.2.0/src/connection.js:20:12)
    at Socket.emit (events.js:160:13)
    at addChunk (_stream_readable.js:269:12)
    at readableAddChunk (_stream_readable.js:256:11)
    at Socket.Readable.push (_stream_readable.js:213:10)
Waiting for the debugger to disconnect...
        
        Failed Step: Vowels in English language are "aeiou".
        Specification: /Users/sswaroop/work/testfolder/js/basic/specs/example.spec:11
        Error Message: Connection closed [127.0.0.1:51190] cause: EOF
        Stacktrace: 
        
        Error Message: write tcp 127.0.0.1:51189->127.0.0.1:51190: use of closed network connection
        Stacktrace: 
        
    Error Message: write tcp 127.0.0.1:51189->127.0.0.1:51190: use of closed network connection
    Stacktrace: 
    
  Error Message: write tcp 127.0.0.1:51189->127.0.0.1:51190: use of closed network connection
  Stacktrace: 
  

  Error Message: write tcp 127.0.0.1:51189->127.0.0.1:51190: use of closed network connection
  Stacktrace: 
  
Successfully generated html-report to => /Users/sswaroop/work/testfolder/js/basic/reports/html-report/index.html
Specifications:	1 executed	0 passed	1 failed	0 skipped
Scenarios:	1 executed	0 passed	1 failed	0 skipped

Total time taken: 3.425s
Error: Tests failed.
```

**Comment by zabil** [comment]
Not happening anymore
